lib/segment/src/index/query_optimization/rescore_formula/parsed_formula.rs

```diff
@@ -76,11 +77,30 @@ impl VariableId {
     }
 }
 
+#[derive(Debug, Clone, PartialEq)]
+pub enum DateTimeExpression {
+    Constant(DateTimePayloadType),
+    PayloadVariable(JsonPath),
+}
+
+impl FromStr for DateTimeExpression {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        // try as date
+        DateTimePayloadType::from_str(s)
+            .map(DateTimeExpression::Constant)
+            // else as payload key
+            .or_else(|_| JsonPath::from_str(s).map(DateTimeExpression::PayloadVariable))
+            .map_err(|_| format!("Invalid date time expression: {s}"))
+    }
+}
```

Comment:
Since the expression is a string, it will try to parse as datetime first, then as a payload key

Reply:
It is not possible to properly distinguish between the two types, right? Not a fan of this, since it's definitely possible for a user to use date time strings as keys.

Maybe like this:

```json
"history": {
  "2025-03-24": 128,
  "2025-03-23": 110,
  "2025-03-22": 283,
  "2025-03-21": 10,
  "2025-03-20": 123,
  "2025-03-19": 28,
  "2025-03-18": 387
}
```

Reply:
That is a good point, I didn't think of the case where datetime strings are used as keys, but it does make sense

Reply:
Solved by introducing a separate expression: `{ "datetime_key": <payload key> }`

Reply:
Thanks. That works!